game/src/core/components/DrawComponent.java

```diff
@@ -340,148 +320,48 @@ public void animationMap(final Map<String, Animation> animationMap) {
-
-          // add animation-files to new or existing storage map
-          if (storage.containsKey(lastDir)) storage.get(lastDir).add(new SimpleIPath(fileName));
-          else {
-            LinkedList<IPath> list = new LinkedList<>();
-            list.add(new SimpleIPath(fileName));
-            storage.put(lastDir, list);
+    final Map<String, List<IPath>> subdirectoryMap = new HashMap<>();
+
+    // Walk through the (sub)directories of path2 and fill the subdirectoryMap.
+    Files.walkFileTree(
+        path2,
+        new SimpleFileVisitor<>() {
+          @Override
+          public FileVisitResult visitFile(final Path file, final BasicFileAttributes attrs) {
+            if (Files.isRegularFile(file)) {
+              subdirectoryMap
+                  .computeIfAbsent(
+                      file.getParent().getFileName().toString(), k -> new ArrayList<>())
+                  .add(new SimpleIPath(file.toString()));
+            }
+            return FileVisitResult.CONTINUE;
           }
-        }
-      }
-    }
+        });
 
-    // sort the files in lexicographic order (like the most os)
-    // animations will be played in order
-    storage.values().forEach(x -> x.sort(Comparator.comparing(IPath::pathString)));
-    // create animations
-    storage.forEach(
-        (name, textureSet) -> animationMap.put(name, Animation.fromCollection(textureSet)));
-    jar.close();
+    // A Map with sorted values (IPath lists) in natural string order (ascending)
+    animationMap =
+        subdirectoryMap.entrySet().stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, DrawComponent::getAnimationFromMapEntry));
   }
 
   /**
-   * Load animations if the game is running in the IDE (or over the shell).
-   *
-   * @param path Path to the animations.
+   * Returns an Animation with sorted paths from the given {@code Map.Entry<String, List<IPath>> x}.
    */
-  private void loadAnimationsFromIDE(final IPath path) {
-    URL url = DrawComponent.class.getResource("/" + path.pathString());
-    if (url != null) {
-      try {
-        try {
-          File apps = new File(url.toURI());
-          animationMap =
-              Arrays.stream(Objects.requireNonNull(apps.listFiles()))
-                  .filter(File::isDirectory)
-                  .collect(Collectors.toMap(File::getName, DrawComponent::allFilesFromDirectory));
-        } catch (IllegalArgumentException e) {
-          LOGGER.log(
-              CustomLogLevel.ERROR, "Could not load animations from directory: " + url.toURI(), e);
-        }
-      } catch (URISyntaxException e) {
-        LOGGER.log(CustomLogLevel.ERROR, "Could not load animations from directory", e);
-      }
-    }
+  private static Animation getAnimationFromMapEntry(Map.Entry<String, List<IPath>> x) {
+    return Animation.fromCollection(
+        x.getValue().stream().sorted(Comparator.comparing(IPath::pathString)).toList());
```

Comment:
Hier passieren zwei Dinge: (a) wird eine Liste mit Pfaden alphabetisch sortiert, und (b) wird aus der Ergebnisliste eine Animation gebaut.

Frage 1: Warum muss ich sortieren? Sollte die Animation nicht auch so klarkommen oder das selbst erledigen? Wir übergeben ihr ja quasi eine Liste von Strings, und wenn die Animation das gern sortiert haben will, soll sie das doch bitte selbst erledigen?!

Frage 2: Wenn wir dennoch hier an dieser Stelle unbedingt eine sortierte Liste haben müssen: Warum nimmst Du dann nicht eine passende Datenstruktur, die das Sortieren selbst erledigt? Wenn ich das richtig sehe, wäre eine SortedSet eigentlich die passende Option - da wir die Sachen aus dem Dateisystem holen, kann es eh keine doppelten Einträge geben (Set-Eigenschaft). Freilich müsste man dann Animation noch anpassen (zumindest eine neue Wrapper-Methode spendieren).

Reply:
Zu Frage (1): Vorher im Code wurde auch sortiert ... Ich denke, weil in einer bestimmten Reihenfolge "gezeichnet" werden sollte.

Zu Frage (2): Das kann ich noch machen später (sortierte Datenstruktur) ... dann wird die Änderung nur etwas größer. ... Ich hatte mich bisher darauf fokussiert, dass es funktioniert, und die Änderungen nicht zu ausufernd werden. Danke erst mal.

Reply:
(2) Ok, Dein Ansatz ist also, erstmal "Pfusch am Bau" einzuführen, und das dass evtl. später nochmal zu ersetzen? Der betreffende Code ist neu in diesem PR (zumindest ist er grün dargestellt) - warum dann nicht ordentlich arbeiten?

Reply:
> (2) Ok, Dein Ansatz ist also, erstmal "Pfusch am Bau" einzuführen, und das dass evtl. später nochmal zu ersetzen? Der betreffende Code ist neu in diesem PR (zumindest ist er grün dargestellt) - warum dann nicht ordentlich arbeiten?

"Pfusch am Bau" würde ich es nicht direkt nennen, ich habe versucht, das Bestehende zu übernehmen und es nicht weiter zu hinterfragen. Aber ich schaue noch mal, ob man die `animationMap` einfach ändern könnte. Danke.

Reply:
done in [18abe0d](https://github.com/Dungeon-CampusMinden/Dungeon/pull/1366/commits/18abe0d56fa6b873940cd09593f95a82de86f2ba)

Reply:
"Pfusch am Bau" ist für mich, wenn Du neuen Code hinzufügst, von dem Du bereits jetzt schon weisst, dass Du ihn nochmal anpacken musst. Ich meinte nicht, direkt in `Animation` oder so Änderungen in diesem PR zu machen, sondern die Sache mit der Liste vs. sortierte Menge.